src/query/src/sql.rs

```diff
@@ -606,6 +618,54 @@ pub fn show_create_table(
     Ok(Output::new_with_record_batches(records))
 }
 
+pub fn show_create_flow(
+    flow_name: ObjectName,
+    flow_val: FlowInfoValue,
+    query_ctx: QueryContextRef,
+) -> Result<Output> {
+    let mut parser = Parser::new(query_ctx.sql_dialect())
+        .with_options(ParserOptions::new().with_trailing_commas(true))
+        .try_with_sql(flow_val.raw_sql())
+        .context(SyntaxSnafu)
+        .context(error::SqlSnafu)?;
+
+    let query = Box::new(
+        parser
+            .parse_query()
+            .context(SyntaxSnafu)
+            .context(error::SqlSnafu)?,
+    );
```

Comment:
We should use `ParserContext` to parse this for consistency. So we don't have to repeat the code creating the parser many times. But we might add a new method to construct the `ParserContext` first (e.g. `ParserContext::new`) and then call `ParserContext::parse_query()`.

@discord9 @WenyXu Do we have a better way to get the query statement? We store the raw sql in the flow value so we have to parse it again in show create flow.